### ISSUE TITLE
webassembly

### ISSUE THREAD

**Issue by pannous** [leafo/moonscript#333]
now that lua runs natively inside of modern browsers (via https://github.com/vvanders/wasm_lua), so should moonscript (theoretically). are there any plans to bundle moonscript into one moonscript.wasm file?

**Comment by CatLover91** [comment]
Wouldn't this just be a case of bundling moonscript on top of the lua wasm? I think having a straight-moonscript wasm would require converting moonscript into a standalone package.

That is a bigger issue, but admittedly it would be nice if I could just `sudo apt install moonscript` on whatever vps I am using.

**Comment by RyanSquared** [comment]
> Wouldn't this just be a case of bundling moonscript on top of the lua wasm?

There'd have to be some changes made (such as, instead of using a filesystem interface, using a DOM interface [if that's possible]); however, it is definitely doable.

> admittedly it would be nice if I could just sudo apt install moonscript on whatever vps I am using.

That depends a lot on the distribution maintainers. `luarocks` is often much more available and will always pull the same version of MoonScript, no matter what system you're on. And, because MoonScript is compatible with all Lua versions (reasonably, 5.1+), it's possible to run on almost any distributed Lua version.

**Comment by CatLover91** [comment]
> That depends a lot on the distribution maintainers. luarocks is often much more available and will always pull the same version of MoonScript, no matter what system you're on. And, because MoonScript is compatible with all Lua versions (reasonably, 5.1+), it's possible to run on almost any distributed Lua version.
That is true, and I did not think of the delay of package maintenance creating poor assumptions on Lua and Moonscript versioning.

**Comment by RyanSquared** [comment]
> That is true, and I did not think of the delay of package maintenance creating poor assumptions on Lua and Moonscript versioning.
 
I hate to say it - package maintenance for Lua programs is **absolutely awful**. Often, you'll only get lua (and *maybe* luarocks) as long as the version is a year old and battle-tested, and have to install all your Lua modules locally (I prefer to use `--local` as well, but that's a habit from attempting to separate *my* packages from *system* packages).
